### PR TITLE
[BREAKING CHANGE]: Move Gatsby config files to Typescript

### PR DESCRIPTION
**Pull request by tlgimenes**
## What's the purpose of this pull request?
As described in the title, moves gatsby connfig files to TS. This PR is a cherry pick of the awesome work done by @victorhmp at #373 

## Checklist
- [x] CHANGELOG entry added

### CODE REVIEW

**Review by danzanzini — APPROVED** [review]

File: gatsby-ssr.tsx [hunk 1]
```diff
@@ -47,15 +49,15 @@ export const onRenderBody = ({ setHeadComponents }) => {
...
+export const onPreRenderHTML: GatsbySSR['onPreRenderHTML'] = ({
   getHeadComponents,
   replaceHeadComponents,
 }) => {
   if (process.env.NODE_ENV !== 'production') {
     return
   }
 
-  const transformedHeadComponents = getHeadComponents().map((node) => {
+  const transformedHeadComponents = getHeadComponents().map((node: any) => {
```
[the remark this answers is not in the document]
Comment (tlgimenes) [reply gatsby-ssr.tsx:60]: using unknown I get these errors. However, I'll try to fix this type
image

[the remark this answers is not in the document]
Comment (igorbrasileiro) [reply gatsby-ssr.tsx]: The gatsby documentation says it's a ReactNode

File: gatsby-ssr.tsx [hunk 2]
```diff
@@ -47,34 +63,36 @@ export const onRenderBody = ({ setHeadComponents }) => {
  * A workaround described in https://github.com/gatsbyjs/gatsby/issues/1526 is
  * implemented below
  */
-export const onPreRenderHTML = ({
+export const onPreRenderHTML: GatsbySSR['onPreRenderHTML'] = ({
   getHeadComponents,
   replaceHeadComponents,
 }) => {
```
Comment (igorbrasileiro) [inline gatsby-ssr.tsx:69]: If you type this parameter you will receive the return type from getHeadComponents

**Review by igorbrasileiro — APPROVED** [review]

**Review by filipewl — APPROVED** [review]
LGTM!

Just as a reminder, there were some suggestions from @igorbrasileiro for some of these files from #207 that we may want to tackle now or later:
- https://github.com/vtex-sites/base.store/pull/207#discussion_r842685289
- https://github.com/vtex-sites/base.store/pull/207#discussion_r842692152
- https://github.com/vtex-sites/base.store/pull/207#discussion_r842692404
- https://github.com/vtex-sites/base.store/pull/207#discussion_r842693497
- https://github.com/vtex-sites/base.store/pull/207#discussion_r842693679

File: gatsby-ssr.tsx (hunk 2 above)
Comment (tlgimenes) [reply gatsby-ssr.tsx:70]: I think it's working now. Can you re-check?

**Pull request merged**